syntax = "proto3";

service FireflyService {
  rpc SendPhases (PhasesRequest) returns (PhasesResponse) {}
  rpc GetPosition (PositionRequest) returns (PositionResponse) {}
}

message Phase {
  int32 x = 1;
  int32 y = 2;
  double phase = 3;
}

message PhasesRequest {
  int32 x = 1;
  int32 y = 2;
  repeated Phase phases = 3;
}

message PhasesResponse {
  repeated Phase phases = 1;
}

message PositionRequest {}

message PositionResponse {
  int32 x = 1;
  int32 y = 2;
}
